lib/iris/tests/unit/coords/test_AuxCoord.py

```diff
@@ -0,0 +1,764 @@
+                bds = np.asarray(self.bds_real, dtype=dtype)
+                if points_type_name == 'lazy':
+                    pts = da.from_array(pts, pts.shape)
+                if bounds_type_name == 'lazy':
+                    bds = da.from_array(bds, bds.shape)
+                elif bounds_type_name == 'no':
+                    bds = None
+                coord = AuxCoord(pts, bounds=bds)
+                result = (coord, points_type_name, bounds_type_name)
+                if _DEBUG_ALL_TYPES:
+                    msg = '\nTest result type {}({} {})'
+                    print(msg.format(coord.dtype, result[1], result[2]))
+                yield result
+
+
+class Test__getitem__(tests.IrisTest):
+    # Test for AuxCoord indexing with various types of points and bounds.
+    def setUp(self):
+        setup_test_arrays(self)
+
+    def test_partial_slice_data_copy(self):
+        parent_coord = AuxCoord([1., 2., 3.])
+        sub_coord = parent_coord[:1]
+        values_before_change = sub_coord.points.copy()
+        parent_coord.points[:] = -999.9
+        self.assertArrayEqual(sub_coord.points, values_before_change)
+
+    def test_full_slice_data_copy(self):
+        parent_coord = AuxCoord([1., 2., 3.])
+        sub_coord = parent_coord[:]
+        values_before_change = sub_coord.points.copy()
+        parent_coord.points[:] = -999.9
+        self.assertArrayEqual(sub_coord.points, values_before_change)
+
+    def test_dtypes(self):
+        # Index coords with all combinations of real+lazy points+bounds, and
+        # either an int or floating dtype.
+        # Check that dtypes are preserved in all cases, taking dtypes directly
+        # from the core points and bounds arrays (as we have no masking).
+        for (main_coord, points_type_name, bounds_type_name) in \
+                coords_all_dtypes_and_lazynesses(self):
+
+            sub_coord = main_coord[:2, 1]
+
+            coord_dtype = main_coord.dtype
+            msg = ('Indexing main_coord of dtype {} '
+                   'with {} points and {} bounds '
+                   'changed dtype of {} to {}.')
+
+            sub_points = _core_points(sub_coord)
+            self.assertEqual(
+                sub_points.dtype, coord_dtype,
+                msg.format(coord_dtype,
+                           points_type_name, bounds_type_name,
+                           'points', sub_points.dtype))
+
+            if bounds_type_name is not 'no':
+                sub_bounds = _core_bounds(sub_coord)
+                self.assertEqual(
+                    sub_bounds.dtype, coord_dtype,
+                    msg.format(coord_dtype,
+                               points_type_name, bounds_type_name,
+                               'bounds', sub_points.dtype))
+
+    def test_lazyness(self):
+        # Index coords with all combinations of real+lazy points+bounds, and
+        # either an int or floating dtype.
+        # Check that laziness is preserved in all cases.
```

Comment:
"laziness is preserved" is supposed to mean "lazy stays lazy + real stays real"
I will try + clarify...